### PR TITLE
Do not vendor kdm and load local file as fallback method

### PR DESCRIPTION
**Pull request by StrongMonkey**
Depends on ~https://github.com/rancher/types/pull/1088~ and ~https://github.com/rancher/rke/pull/1910~

### CODE REVIEW

File: pkg/image/export/main.go [hunk 1]
```diff
@@ -57,12 +58,23 @@ func run(systemChartPath string, imagesFromArgs []string) error {
 	if strings.HasPrefix(rancherVersion, "v") {
 		rancherVersion = rancherVersion[1:]
 	}
+
+	// already downloaded in dapper
+	b, err := ioutil.ReadFile("/root/.data.json")
```
Comment (superseb) [inline pkg/image/export/main.go]: This is `/root/data.json` I assume

File: package/Dockerfile [hunk 2]
```diff
@@ -15,7 +15,7 @@ ARG ARCH=amd64
 ARG IMAGE_REPO=rancher
 ARG SYSTEM_CHART_DEFAULT_BRANCH=dev
 # kontainer-driver-metadata branch to be set for specific branch other than dev/master, logic at rancher/rancher/pkg/settings/setting.go  
-ARG RANCHER_METADATA_BRANCH=
+ARG RANCHER_METADATA_BRANCH=dev
```
Comment (superseb) [inline package/Dockerfile]: This was changed in https://github.com/rancher/rancher/commit/5ced4dc03384f44db3682bf0652d57274b90e02f

**Comment by StrongMonkey** [comment]
@superseb Changes updated.

File: package/Dockerfile [hunk 3]
```diff
@@ -66,7 +66,9 @@ RUN curl -sLf ${!TINI_URL} > /usr/bin/tini && \
     curl -sfL ${!ETCD_URL} | tar xvzf - --strip-components=1 -C /usr/bin/ etcd-${CATTLE_ETCD_VERSION}-linux-${ARCH}/etcd etcd-${CATTLE_ETCD_VERSION}-linux-${ARCH}/etcdctl && \
     curl -sLf https://github.com/rancher/telemetry/releases/download/${TELEMETRY_VERSION}/telemetry-${ARCH} > /usr/bin/telemetry && \
     curl -sLf https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/${ARCH}/kubectl > /usr/bin/kubectl && \
-    chmod +x /usr/bin/rancher-helm /usr/bin/tini /usr/bin/telemetry /usr/bin/rancher-tiller /usr/bin/k3s /usr/bin/kubectl
+    chmod +x /usr/bin/rancher-helm /usr/bin/tini /usr/bin/telemetry /usr/bin/rancher-tiller /usr/bin/k3s /usr/bin/kubectl && \
+    mkdir -p /var/lib/rancher/management-state/driver-metadata && \
+    curl -sLf https://releases.rancher.com/kontainer-driver-metadata/${RANCHER_METADATA_BRANCH}/data.json > /var/lib/rancher/management-state/driver-metadata/data.json
```
Comment (superseb) [inline package/Dockerfile]: `/var/lib/rancher` is what we document as mountable location from the host, please use the directory as shown in https://github.com/rancher/rancher/pull/25388/files#diff-fa24c4fe8d6b6653fad727c37ac24b99R33

Comment (StrongMonkey) [reply package/Dockerfile]: ah, ok. The file was historically stored in there but I will move it.

Comment (StrongMonkey) [reply package/Dockerfile]: @superseb updated

**Comment by StrongMonkey** [comment]
@superseb That addon method has been fixed and I did a sanity check and it should be working now.

**Comment by StrongMonkey** [comment]
Address https://github.com/rancher/rancher/issues/25162.

**Pull request merged**